src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -0,0 +1,51 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Common analyzer functionality"""
+
+from operator import attrgetter
+
+from pandas import DataFrame
+
+class Analyzer():
+    """A base class providing common analyzer functionality
+
+    Derived classes must override class attribute `cols`, specifying a
+    list of column names to extract from collected rows of data.
+    """
+    cols = ()
+    def __init__(self):
+        getter = attrgetter(*self.cols)
+        if len(self.cols) > 1:
+            self._row_builder = getter
+        else:
+            self._row_builder = lambda row: (getter(row),)
+        self._rows = []
+        self._data = None
+    def collect(self, *rows):
+        """Collect data from `rows`"""
+        self._rows.extend([self._row_builder(r) for r in rows])
```

Comment:
Sorry bad wording I meant this 
```suggestion
    def collect(self, *rows):
        """Collect data from `rows`"""
        if self._rows is None:
            raise CollectionClosed()
        self._rows.extend([self._row_builder(r) for r in rows])
```